lib/run_part.c

```diff
@@ -80,7 +80,7 @@ int run_parts (const char *directory, const char *name, const char *action)
 			return (1);
 		}
 
-		if (S_ISREG (sb.st_mode) || S_ISLNK (sb.st_mode)) {
```

Comment:
LGTM.  At least I couldn't make it return true with some test.  Even a broken link or a link pointing to itself won't trigger this, because stat(2) will just fail.

For patch 2:

```
Reviewed-by: Alejandro Colomar <alx@kernel.org>
```